pkg/operator/sync_openshiftcontrollermanager_v311_00_test.go

```diff
@@ -834,58 +819,120 @@ func TestDeploymentWithProxy(t *testing.T) {
+				c.Status.HTTPProxy = c.Spec.HTTPProxy
+				c.Status.HTTPSProxy = c.Spec.HTTPSProxy
+				return c
+			},
+		},
 	}
 
-	foundNoProxy := false
-	foundHTTPProxy := false
-	foundHTTPSProxy := false
-	for _, c := range ds.Spec.Template.Spec.Containers {
-		for _, e := range c.Env {
-			switch e.Name {
-			case "NO_PROXY":
-				if e.Value == proxyConfig.Status.NoProxy {
-					foundNoProxy = true
-				}
-			case "HTTP_PROXY":
-				if e.Value == proxyConfig.Status.HTTPProxy {
-					foundHTTPProxy = true
+	var currentProxyConfig *configv1.Proxy
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// Update proxy config on demand.
+			if tt.newProxyConfig != nil {
+				if c := tt.newProxyConfig(currentProxyConfig); c != nil {
+					if currentProxyConfig == nil {
+						indexer.Add(c)
+					} else {
+						indexer.Update(c)
+					}
+					currentProxyConfig = c
 				}
-			case "HTTPS_PROXY":
-				if e.Value == proxyConfig.Status.HTTPSProxy {
-					foundHTTPSProxy = true
+			}
+
+			// Sync.
+			ds, rcBool, err := manageOpenShiftControllerManagerDeployment_v311_00_to_latest(
+				deployClient,
+				countNodes,
+				workloadcontroller.EnsureAtMostOnePodPerNode,
+				recorder,
+				operatorConfig,
+				"my.co/repo/img:latest",
+				operatorConfig.Status.Generations,
+				proxyLister,
+				specAnnotations,
+			)
+
+			// Check results.
+			if err != nil {
+				t.Fatalf("unexpected error: %s", err.Error())
+			}
+			if !rcBool {
+				t.Fatal("apply deployment does not think a changes was made")
+			}
+			if ds == nil {
+				t.Fatalf("nil deployment returned")
+			}
+
+			foundNoProxy := false
+			foundHTTPProxy := false
+			foundHTTPSProxy := false
+			foundPodName := false
+			for _, c := range ds.Spec.Template.Spec.Containers {
+				for _, e := range c.Env {
+					switch e.Name {
+					case "NO_PROXY":
+						if e.Value == currentProxyConfig.Status.NoProxy {
+							foundNoProxy = true
+						}
+					case "HTTP_PROXY":
+						if e.Value == currentProxyConfig.Status.HTTPProxy {
+							foundHTTPProxy = true
+						}
+					case "HTTPS_PROXY":
+						if e.Value == currentProxyConfig.Status.HTTPSProxy {
+							foundHTTPSProxy = true
+						}
+					case "POD_NAME":
+						foundPodName = true
+					}
 				}
 			}
-		}
-	}
 
-	if !foundNoProxy {
-		t.Fatalf("NO_PROXY not found: %#v", ds.Spec.Template.Spec.Containers)
-	}
-	if !foundHTTPProxy {
-		t.Fatalf("HTTP_PROXY not found: %#v", ds.Spec.Template.Spec.Containers)
-	}
-	if !foundHTTPSProxy {
-		t.Fatalf("HTTPS_PROXY not found: %#v", ds.Spec.Template.Spec.Containers)
+			if !foundNoProxy {
+				t.Fatalf("NO_PROXY not found: %#v", ds.Spec.Template.Spec.Containers)
+			}
+			if !foundHTTPProxy {
+				t.Fatalf("HTTP_PROXY not found: %#v", ds.Spec.Template.Spec.Containers)
+			}
+			if !foundHTTPSProxy {
+				t.Fatalf("HTTPS_PROXY not found: %#v", ds.Spec.Template.Spec.Containers)
+			}
+			if !foundPodName {
+				t.Fatalf("POD_NAME not found: %#v", ds.Spec.Template.Spec.Containers)
+			}
+		})
```

Comment:
I think we can just call manageOpenShiftControllerManagerDeployment_v311_00_to_latest here again

Reply:
This is wrong - sorry for the misguidance. We do not use the cluster state when applying the deployment: https://github.com/openshift/cluster-openshift-controller-manager-operator/blob/83746c46b6483bafad6a82832bc70b7f612a4789/vendor/github.com/openshift/library-go/pkg/operator/resource/resourceapply/apps.go#L140